### PR TITLE
[Merged by Bors] - feat: port NumberTheory.Zsqrtd.GaussianInt

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
https://leanprover-community.github.io/mathlib-port-status/file/number_theory/zsqrtd/gaussian_int
seems to indicate that this has all dependencies ported, but it seems like it needs
https://leanprover-community.github.io/mathlib-port-status/file/number_theory/legendre_symbol/quadratic_reciprocity,
which at the time of writing has
Dependencies 16 + 2 + 1008

**Comment by collares** [comment]
That's just because leanprover-community/mathlib#1919e hasn't been processed by mathport yet, I think.

**Comment by mo271** [comment]
ok, cool, there should be only one missing dependency, namely `number_theory/zsqrtd/quadratic_reciprocity.lean` for GaussianInt,  I suppose

File: Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean [hunk 1]
```diff
@@ -77,99 +77,99 @@ end
...
-  rw [← to_complex_zero, to_complex_inj]
-#align gaussian_int.to_complex_eq_zero GaussianInt.to_complex_eq_zero
+theorem toComplex_eq_zero {x : ℤ[i]} : (x : ℂ) = 0 ↔ x = 0 := by
+  rw [← toComplex_zero, toComplex_inj]
+#align gaussian_int.to_complex_eq_zero GaussianInt.toComplex_eq_zero
 
 @[simp]
-theorem nat_cast_real_norm (x : ℤ[i]) : (x.norm : ℝ) = (x : ℂ).normSq := by
-  rw [Zsqrtd.norm, norm_sq] <;> simp
+theorem nat_cast_real_norm (x : ℤ[i]) : (x.norm : ℝ) = Complex.normSq (x : ℂ) := by
```
Comment (urkud) [inline Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean:154]: There are no `nat`s in this theorem. LHS uses `Int.cast`, not `Nat.cast`.
```suggestion
theorem int_cast_real_norm (x : ℤ[i]) : (x.norm : ℝ) = Complex.normSq (x : ℂ) := by
```

File: Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean [hunk 2]
```diff
@@ -77,99 +77,99 @@ end
...
+  rw [Zsqrtd.norm, normSq]; simp
 #align gaussian_int.nat_cast_real_norm GaussianInt.nat_cast_real_norm
```
Comment (urkud) [inline Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean:156]:
```suggestion
#align gaussian_int.nat_cast_real_norm GaussianInt.int_cast_real_norm
```

File: Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean [hunk 3]
```diff
@@ -77,99 +77,99 @@ end
...
 
 @[simp]
-theorem nat_cast_complex_norm (x : ℤ[i]) : (x.norm : ℂ) = (x : ℂ).normSq := by
-  cases x <;> rw [Zsqrtd.norm, norm_sq] <;> simp
+theorem nat_cast_complex_norm (x : ℤ[i]) : (x.norm : ℂ) = Complex.normSq (x : ℂ) := by
```
Comment (urkud) [inline Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean:159]: Same here.

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

**Pull request closed without merging**